fieldtofile/fieldtofile-service.ts

```diff
@@ -12,70 +19,18 @@ export async function executeFieldToFile(environment: PH.ServiceTask.IServiceTas
 
-  // Connect to your database
   try {
-
-    let baseDir = null;
-    if (fs.existsSync(process.cwd() + "/roxtra-config-custom.json")) {
-      const customConfigString = fs.readFileSync(process.cwd() + "/roxtra-config-custom.json", "utf8");
-      const configObj = JSON.parse(customConfigString);
-      if (configObj["Filestore"] != null && configObj["Filestore"]["baseDir"] != null && configObj["Filestore"]["baseDir"] !== "") {
-        baseDir = configObj["Filestore"]["baseDir"];
-      }
-    }
-
-    if (baseDir == null && fs.existsSync(process.cwd() + "/roxtra-config.json")) {
-      const customConfigString = fs.readFileSync(process.cwd() + "/roxtra-config.json", "utf8");
-      const configObj = JSON.parse(customConfigString);
-      if (configObj["Filestore"] != null && configObj["Filestore"]["baseDir"] != null && configObj["Filestore"]["baseDir"] !== "") {
-        baseDir = configObj["Filestore"]["baseDir"];
-      }
-    }
-
-    if (baseDir == null) {
-      return false;
-    }
-
-    const fileUrls: string[] = (environment.instanceDetails.extras.fieldContents[sourceField] as PH.Data.IFieldValue).value as string[];
+    const fileUrls: string[] = (environment.fieldContents[sourceField] as PH.Data.IFieldValue).value as string[];
     for (const fileUrl of fileUrls) {
-      if (fileUrl != null && fileUrl !== "") {
-
-        const url = fileUrl;
-        const downloadRoutPrefix = "eformulare/files/";
-        const downloadRoutePrefixIdx: number = url.indexOf(downloadRoutPrefix);
-        const draftFileKey: string = url.substr(downloadRoutePrefixIdx + downloadRoutPrefix.length);
-        const filePath = baseDir + "/" + draftFileKey;
-
+      if (fileUrl) {
         const filename = fileUrl.split("/").last();
+        const relativePath = fileUrl.replace(getBackendUrl() + "/files/", "");
 
-        // eslint-disable-next-line capitalized-comments
-        /* let dir = "tmp";
+        const fileData = await getAttachedFile(environment, relativePath);
 
-        if (!fs.existsSync(dir)) {
-          fs.mkdirSync(dir);
-        }*/
-        // + environment.instanceDetails.instanceId + "/"
-        fs.createReadStream(filePath).pipe(fs.createWriteStream(targetPath + "/" + filename));
-        // eslint-disable-next-line capitalized-comments
-        // let promise = new Promise<void>((resolve, reject) => {
-        //   try {
-        //     http.get(fileUrl, (response: any) => {
-        //       let filename = fileUrl.split("/").last();
-        //       console.log(filename);
-        //       console.log(targetPath + "/" + filename);
-        //       const file = fs.createWriteStream(targetPath + "/" + filename);
-        //       response.pipe(file);
-        //       resolve();
-        //     });
-        //   } catch (ex) {
-        //     reject();
-        //     return false;
-        //   }
-        // });
-        // await promise;
+        fs.writeFileSync(targetPath + "/" + filename, fileData);
```

Comment:
Maybe another option - why don't you just download the file from the original URL and save it to the target path?

Reply:
The files are now downloaded from the download link.